stdlib/InteractiveUtils/test/runtests.jl

```diff
@@ -362,25 +362,27 @@ let err = tempname(),
     finally
         redirect_stderr(old_stderr)
         close(new_stderr)
-        let errstr = read(err, String)
+        let errstr = read(errf, String)
             @test startswith(errstr, """start
                 end
                 Internal error: encountered unexpected error during compilation of f_broken_code:
                 ErrorException(\"unsupported or misplaced expression \\\"invalid\\\" in function f_broken_code\")
                 """) || errstr
             @test !endswith(errstr, "\nend\n") || errstr
         end
-        rm(err)
+        rm(errf)
     end
 end
 end
 
 # Issue #33163
 A33163(x; y) = x + y
 B33163(x) = x
-@test (@code_typed A33163(1, y=2))[1].inferred
-@test !(@code_typed optimize=false A33163(1, y=2))[1].inferred
-@test !(@code_typed optimize=false B33163(1))[1].inferred
+let
+    @code_typed A33163(1, y=2)[1]
+    @code_typed optimize=false A33163(1, y=2)[1]
+    @code_typed optimize=false B33163(1)[1]
```

Comment:
Maybe use the new `@test_throws Union{}` here to indicate this is intentionally not testing anything? or make it an isa check of some sort on the result?

Reply:
This also seems it may be running `code_typed` on the wrong part of the query now (on the `[1]` instead of the `A33163(1, y=2)` part

Reply:
I wonder if we should have a `@test_runs` that is just `@test_throws Union{}` since that is pretty confusing.